### ISSUE TITLE
When I try to install it, the prompt 'Could not extract app extract' Unable to install

### ISSUE THREAD

**Issue by kanglyh** [PaulLereverend/NextcloudExtract#8]
When I try to install it, the prompt 'Could not extract app extract' Unable to install

**Comment by PaulLereverend** [comment]
Hello, 

- What version of nextcloud do you use ? 
- Are you able to install other apps ? 
- Did you tried to delete the "extract" folder inside nextcloud/apps/ and reinstall ?

**Comment by kanglyh** [comment]
Hello, thank you for giving me a reply
Nextcloud version is 15.0.7
I can install other apps
I can't find the "extract" folder, so I can't delete it

**Comment by PaulLereverend** [comment]
What OS is running your NC instance ?
Which version of php ?
Do you have something related to this on NC logs ?

**Comment by kanglyh** [comment]
OS is Ubuntu 16.04.3 
php version is 7.2
I am trying to modify the database to solve the problem.

**Comment by PaulLereverend** [comment]
I don't think it is related to the app. Try to install the new version (1.2.0), this may solve the problem.

**Comment by kanglyh** [comment]
[settings] Error: Exception: Could not extract app extract at <<closure>>

0. /www/wwwroot/pan/settings/Controller/AppSettingsController.php line 438
   downloadApp("extract")
1. /www/wwwroot/pan/lib/private/AppFramework/Http/Dispatcher.php line 166
   enableApps(["extract"], [])
2. /www/wwwroot/pan/lib/private/AppFramework/Http/Dispatcher.php line 99
   executeController(OC\Settings\Cont ... {}, "enableApps")
3. /www/wwwroot/pan/lib/private/AppFramework/App.php line 118
   dispatch(OC\Settings\Cont ... {}, "enableApps")
4. /www/wwwroot/pan/lib/private/AppFramework/Routing/RouteActionHandler.php line 47
   main("OC\\Settings\\C ... r", "enableApps", OC\AppFramework\ ... {}, {_route: "settin ... "})
5. <<closure>>
   __invoke({_route: "settin ... "})
6. /www/wwwroot/pan/lib/private/Route/Router.php line 297
   call_user_func(OC\AppFramework\ ... {}, {_route: "settin ... "})
7. /www/wwwroot/pan/lib/base.php line 987
   match("/settings/apps/enable")
8. /www/wwwroot/pan/index.php line 42
   handleRequest()

POST /index.php/settings/apps/enable
from 103.117.103.206 by klyh at 2019-04-30T12:10:32+00:00

**Comment by kanglyh** [comment]
[settings] Error: Exception: Could not extract app extract at <<closure>>

0. /www/wwwroot/pan/settings/Controller/AppSettingsController.php line 438
   downloadApp("extract")
1. /www/wwwroot/pan/lib/private/AppFramework/Http/Dispatcher.php line 166
   enableApps(["extract"], [])
2. /www/wwwroot/pan/lib/private/AppFramework/Http/Dispatcher.php line 99
   executeController(OC\Settings\Cont ... {}, "enableApps")
3. /www/wwwroot/pan/lib/private/AppFramework/App.php line 118
   dispatch(OC\Settings\Cont ... {}, "enableApps")
4. /www/wwwroot/pan/lib/private/AppFramework/Routing/RouteActionHandler.php line 47
   main("OC\\Settings\\C ... r", "enableApps", OC\AppFramework\ ... {}, {_route: "settin ... "})
5. <<closure>>
   __invoke({_route: "settin ... "})
6. /www/wwwroot/pan/lib/private/Route/Router.php line 297
   call_user_func(OC\AppFramework\ ... {}, {_route: "settin ... "})
7. /www/wwwroot/pan/lib/base.php line 987
   match("/settings/apps/enable")
8. /www/wwwroot/pan/index.php line 42
   handleRequest()

POST /index.php/settings/apps/enable
from ****** by klyh at 2019-04-30T12:10:32+00:00

**Comment by kanglyh** [comment]
[settings] Error: Exception: Could not extract app extract at <<closure>>

0. /www/wwwroot/pan/settings/Controller/AppSettingsController.php line 438
   downloadApp("extract")
1. /www/wwwroot/pan/lib/private/AppFramework/Http/Dispatcher.php line 166
   enableApps(["extract"], [])
2. /www/wwwroot/pan/lib/private/AppFramework/Http/Dispatcher.php line 99
   executeController(OC\Settings\Cont ... {}, "enableApps")
3. /www/wwwroot/pan/lib/private/AppFramework/App.php line 118
   dispatch(OC\Settings\Cont ... {}, "enableApps")
4. /www/wwwroot/pan/lib/private/AppFramework/Routing/RouteActionHandler.php line 47
   main("OC\\Settings\\C ... r", "enableApps", OC\AppFramework\ ... {}, {_route: "settin ... "})
5. <<closure>>
   __invoke({_route: "settin ... "})
6. /www/wwwroot/pan/lib/private/Route/Router.php line 297
   call_user_func(OC\AppFramework\ ... {}, {_route: "settin ... "})
7. /www/wwwroot/pan/lib/base.php line 987
   match("/settings/apps/enable")
8. /www/wwwroot/pan/index.php line 42
   handleRequest()

POST /index.php/settings/apps/enable
from ****** by ******* at 2019-04-30T12:10:32+00:00

**Comment by Schmuuu** [comment]
In my opinion, this does not have anything to do with the "Extract" app. There is a mis-configuration on your server and you should rather contact the NC forum as soon as it is available again.

One thing you can check before posting in the forum is, if you have **php-zip** installed and enabled.

**Comment by PaulLereverend** [comment]
@Schmuuu is right, there is nothing related to the code of this app in this error. 

You could try to extract this into nextcloud/apps and enable it from from your panel https://github.com/PaulLereverend/NextcloudExtract/releases/download/1.2.0/extract.tar.gz
